jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/Schema.java

```diff
@@ -28,13 +28,13 @@
 
     private final URI id;
     private final JsonNode content;
-    private final JsonNode parentContent;
+    private final Schema parent;
     private JType javaType;
 
-    public Schema(URI id, JsonNode content, JsonNode parentContent) {
+    public Schema(URI id, JsonNode content, Schema parent) {
         this.id = id;
         this.content = content;
-        this.parentContent = parentContent;
+        this.parent = parent != null ? parent : this;
```

Comment:
Could you help me understand why this is needed? If there is no parent, why is the parent 'this'?

Reply:
This is to mimic the logic you used before. Previously, the _parent_ in `parentContent` didn't really mean _parent or null if none_, but rather _parent or self aka "root"_. I adopted this style, because it meant minimal code changes and frees the caller of `getParent` from performing null-checks. I have no personal opinion whether `null` or `this` is better or worse.